pkg/datarepair/checker/checker.go

```diff
@@ -51,3 +50,105 @@ func (c Config) Run(ctx context.Context, server *provider.Provider) (err error)
 
 	return server.Run(ctx)
 }
+
+// Checker contains the information needed to do checks for missing pieces
+type Checker struct {
+	params      *pb.IdentifyRequest
+	pointerdb   storage.KeyValueStore
+	repairQueue *queue.Queue
+	overlay     pb.OverlayServer
```

Comment:
Shouldn't we actually use the `overlay.Client` interface instead of the "server" one? The repair checker is a client to the overlay cache, isn't it?

Reply:
JT said since the checker will run on the server, the overlay server works here. It also allows us to test with the mock cache.